pootle/apps/virtualfolder/receivers.py

```diff
@@ -19,85 +20,67 @@ def update_vfolder_tree(vf, store):
     """
     # Create missing VirtualFolderTreeItem tree structure for affected Stores
     # after adding or unobsoleting Units.
-    vfolder_treeitem, created = (
-        VirtualFolderTreeItem.objects.get_or_create(
-            directory=store.parent, vfolder=vf))
-
+    created = False
+    try:
+        vfolder_treeitem = (
+            VirtualFolderTreeItem.objects.get(
+                directory=store.parent, vfolder=vf))
+    except VirtualFolderTreeItem.DoesNotExist:
+        parts = split_pootle_path(store.parent.pootle_path)
+        path_parts = ["", parts[0], parts[1], vf.name]
```

Comment:
You are assuming that vfolder name is always just after language/project, but the location could have been deeper.